[quant][pyper] Make offsets an optional paramter in the qembedding_bag op

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#42924 [quant][pyper] Make offsets an optional paramter in the qembedding_bag op**

Summary:
offsets is an optional paramter in the python module currently. So we update the operator to follow suit
in order to avoid bad optional access

Test Plan:
python test/test_quantization.py TestQuantizeDynamicJitOps.test_embedding_bag

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23081152](https://our.internmc.facebook.com/intern/diff/D23081152)

supriyar committed:
[quant][pyper] Make offsets an optional paramter in the qembedding_bag op
Summary:
offsets is an optional paramter in the python module currently. So we update the operator to follow suit
in order to avoid bad optional access

Test Plan:
python test/test_quantization.py TestQuantizeDynamicJitOps.test_embedding_bag

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

radkris-git commented on aten/src/ATen/native/quantized/cpu/qembeddingbag.cpp:
```diff
@@ -112,15 +115,19 @@ Tensor embedding_bag_byte_rowwise_offsets(
     const c10::optional<Tensor>& per_sample_weights_,
     const c10::optional<Tensor>& compressed_indices_mapping,
     bool include_last_offset) {
+  TORCH_CHECK(offsets_in.has_value(), "qembedding_bag expects offsets to be set");
```
s/qembedding_bag /embedding_bag_4bit_rowwise_offsets/g :)

radkris-git commented on aten/src/ATen/native/quantized/cpu/qembeddingbag.cpp:
```diff
@@ -14,14 +14,17 @@ namespace native {
     bool include_last_offset) {
   TORCH_CHECK(weight.scalar_type() == at::kByte);
   TORCH_CHECK(weight.ndimension() == 2);
+  TORCH_CHECK(offsets_in.has_value(), "qembedding_bag expects offsets to be set");
```
s/qembedding_bag /embedding_bag_byte_rowwise_offsets/g :)

supriyar committed:
Update on "[quant][pyper] Make offsets an optional paramter in the qembedding_bag op"


Summary:
offsets is an optional paramter in the python module currently. So we update the operator to follow suit
in order to avoid bad optional access

Test Plan:
python test/test_quantization.py TestQuantizeDynamicJitOps.test_embedding_bag

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23081152](https://our.internmc.facebook.com/intern/diff/D23081152)

[ghstack-poisoned]
